src/plugins/controls/public/actions/clear_control_action.test.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { BehaviorSubject } from 'rxjs';
+import { getOptionsListControlFactory } from '../controls/data_controls/options_list_control/get_options_list_control_factory';
+import { getMockedBuildApi, getMockedControlGroupApi } from '../controls/mocks/control_mocks';
+import { ClearControlAction } from './clear_control_action';
+import { CanClearSelections } from '../types';
+
+import type { ViewMode, EmbeddableApiContext } from '@kbn/presentation-publishing';
+import type { OptionsListControlApi } from '../controls/data_controls/options_list_control/types';
+
+const dashboardApi = {
+  viewMode: new BehaviorSubject<ViewMode>('view'),
+};
+const controlGroupApi = getMockedControlGroupApi(dashboardApi, {
+  removePanel: jest.fn(),
+  replacePanel: jest.fn(),
+  addNewPanel: jest.fn(),
+  children$: new BehaviorSubject({}),
+});
+
+let clearControlAction: ClearControlAction;
+let embeddable: Partial<EmbeddableApiContext & CanClearSelections>;
+let optionsListApi: OptionsListControlApi;
```

Comment:
One of the strengths of the new embeddable is that testing can be decouple from implementations.  Since this action is not tied to a specific control but rather any control that implements `CanClearSelections` It may be best to de-couple these tests from options list control, instead using a mock embeddable API.

I suggest removing `optionsListApi`, `embeddable`, and `beforeAll` block. Leaving the following test setup

```
const dashboardApi = {
  viewMode: new BehaviorSubject<ViewMode>('view'),
};
const controlGroupApi = getMockedControlGroupApi(dashboardApi, {
  removePanel: jest.fn(),
  replacePanel: jest.fn(),
  addNewPanel: jest.fn(),
  children$: new BehaviorSubject({}),
});

const clearControlAction = new ClearControlAction();
const hasSelections$: BehaviorSubject<boolean | undefined>;
const controlApi = {
  type: 'test',
  uuid: '1',
  parentApi: controlGroupApi,
  hasSelections$,
  clearSelections: jest.fn(),
}

beforeEach(() => {
  hasSelections$.next(false);
});
```